### PR TITLE
Adding Traffic Violation example

### PR DESCRIPTION
**Pull request by etirelli**

### CODE REVIEW

**Comment by manstis** [comment]
@etirelli I think you also need to change these files to get your new example included.

https://github.com/kiegroup/kie-wb-playground/blob/master/kie-decision-playground/src/main/assembly/assembly-zip-kie-decision-playground.xml

https://github.com/kiegroup/kie-wb-playground/blob/master/kie-all-playground/src/main/assembly/assembly-zip-kie-all-playground.xml

**Comment by etirelli** [comment]
Thank you for the heads up @manstis !

**Comment by etirelli** [comment]
@manstis @jomarko can you guys please review?

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
@etirelli Both @jomarko and I tried this PR.

Unfortunately the new project does not appear in any of the webapps; the reason being we can only guess at. I did notice that the parent pom of the new project is not consistent with other example projects however changing that locally only led to a heap of other issues... that on face value suggest DMN's dependencies are not defined in `org.kie:kie-bom`.

@adrielparedes and @yurloc have added _new_ projects more recently so they may be best to advise what needs changing in your PR in order for the traffic-violations project to be (a) correctly defined and (b) included in the Business Central distributions.

Sorry, but I'm not 100% familiar with how examples now work and hence it's probably best I defer to others more knowledgeable (I've spent 3 hours this morning trying to fix but to no avail).

**Comment by etirelli** [comment]
@adrielparedes @yurloc can you please help us with this? we need to add this DMN example to the list of examples a user can import. The only thing I did was to export the repository out of my local business central and add it to this repo.

@danielezonca FYI, we might need to add the DMN dependencies somewhere else.

**Comment by danielezonca** [comment]
@etirelli 
What dependencies are you referring to? If you are referring to this project POM I cannot see any pom parent and in any case the GAV is complete so I don't understand how `kie-bom` can be related. Even if we change the project to use `kie-bom` as parent it should not be involved with dependency management if the GAV is fully specified

@manstis 
From what I know all the existing examples are not created with the current BC project "template" but with and old one or maybe manually. I think this is the reason why they are different

**Comment by manstis** [comment]
@danielezonca Simply put building this PR and `kie-wb-distributions` does not result in the new project being available to Business Central. I was trying to work out what may be causing it so noticed the other examples have a different parent to that in Edson's new project. Changing that locally lead to various issues relating to KIE DMN dependency issues. After more fiddling around and no success I thought it best to suggest someone who knows how to add new projects to BC may be better helping Edson.

Not knowing how new examples should _look_ (code/structure-wise) I could only compare Edson's new example to existing and fiddle around trying to make them comparable.. if there's an easier way to get the new project in BC then happy to hear!

**Comment by etirelli** [comment]
@danielezonca @mbiarnes this PR is failing with:

10:31:26 Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.drools:drools-wb-scenario-simulation-editor-api:jar:7.18.0-SNAPSHOT


Any idea why the dependency is not found?

**Comment by danielezonca** [comment]
@etirelli 
TBH I have no idea, I tried locally to build it using the same Maven command
`mvn -B -e -T1C -DskipTests -Dgwt.compiler.skip=true -Dgwt.skipCompilation=true -Denforcer.skip=true -Dcheckstyle.skip=true -Dfindbugs.skip=true -Drevapi.skip=true clean install`
but I have no error.
Maybe we should just ask 
jenkins retest this please
?

@mbiarnes 
Any suggestion?

**Comment by danielezonca** [comment]
@etirelli 
This time failed on backend module instead of API. I don't know.

I looked at the build logs (https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/kie-wb-playground-pullrequests/10/consoleText) and I found something strange:
`Using repository-list.txt from GitHubRepository{owner='kiegroup', name='droolsjbpm-build-bootstrap'},GitBranch{name='master'}.
GitHub repositories that will be cloned and built:
	git://github.com/errai/errai.git:master:master-pr-build
	git://github.com/kiegroup/lienzo-core.git:master:master-pr-build
	git://github.com/kiegroup/lienzo-tests.git:master:master-pr-build
	git://github.com/kiegroup/kie-soup.git:master:master-pr-build
	git://github.com/kiegroup/appformer.git:master:master-pr-build
	git://github.com/kiegroup/droolsjbpm-build-bootstrap.git:master:master-pr-build
	git://github.com/kiegroup/droolsjbpm-knowledge.git:master:master-pr-build
	git://github.com/kiegroup/drlx-parser.git:master:master-pr-build
	git://github.com/kiegroup/drools.git:master:master-pr-build
	git://github.com/kiegroup/optaplanner.git:master:master-pr-build
	git://github.com/kiegroup/jbpm.git:master:master-pr-build
	git://github.com/kiegroup/droolsjbpm-integration.git:master:master-pr-build
	git://github.com/kiegroup/droolsjbpm-tools.git:master:master-pr-build
	git://github.com/kiegroup/kie-uberfire-extensions.git:master:master-pr-build`

This seems the list of repos used for the build. This list doesn't contain `drools-wb` project (I assume no other projects has a dependency on `drools-wb` code except drools-wb itself + now this project).
Btw the message is saying that is using repository-list.txt but that file contains also `drools-wb` so probably the file used is not this 
https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/master/script/repository-list.txt

I assume we have another repository-list.txt file somewhere else to be updated or manage this as exception like it seems is written in some comments here for other project/build
https://github.com/kiegroup/kie-jenkins-scripts/search?q=repository-list.txt&unscoped_q=repository-list.txt

This is everything I understand from the logs, I don't know exactly the rest of the pipeline so @mbiarnes 
i-choose-you

**Comment by porcelli** [comment]
@etirelli you have a dependency on these modules from the drools-wb repository. however, this playground repository is built before [1]. you may need to remove the dependency OR change the build order.

[1] - https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/master/script/repository-list.txt

File: traffic-violation/pom.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.kie.example.traffic</groupId>
```
Comment (adrielparedes) [inline traffic-violation/pom.xml]: playground parent pom missing

```
<parent>
    <groupId>org.kie.workbench.playground</groupId>
    <artifactId>playground-parent</artifactId>
    <version>7.18.0-SNAPSHOT</version>
  </parent>
```

**Comment by porcelli** [comment]
I opened a new one, with some fixes that should work

**Pull request closed without merging**